### DISCUSSION TITLE
Accessing the onRendered paramater

### DISCUSSION THREAD

**Discussion by NicoN13** [anvilistas/tabulator#129] [Q&A] [answered]
I'd like to include Sparklines in my tabulator. From reading the tabulator docs at:
https://tabulator.info/docs/5.5/format#main-contents
It seems we need to access the onRendered call back function to use certain libraries (Sparkline included).

It doesn't seem like onRendered is included in the params:
![image](https://github.com/s-cork/Tabulator/assets/117717063/1336f1d1-5f04-466c-9120-02383e50acfc)

How would you recommend I go about this?

What I've done:

I include the following script in my Native Libraries:
```<script type="text/javascript" src="jquery.sparkline.js"></script>```

I've tried using sparklines in my standard anvil python formatter definition like this:

```
def sparkline_formatter(cell, **params):
  anvil.js.call_js("addSparkLine", cell)
```

And in Native Libraries I have the following function defined:

```
<script>
  function addSparkLine(cell) {
    $(cell.getElement()).sparkline([1,2,3,4], {width: "100%", type: "line"})
  }
</script>
```

**Comment by s-cork** [comment]
Yeah - that seems reasonable
Can you create a minimal example clone link so that I can play with it and add the feature
(obviously the clone link won't work but just what you think should work)

**Reply by NicoN13** [reply]
Thanks @s-cork 

Here's a clone:
https://anvil.works/build#clone:7FJBZRHZ4ENEQWKR=KUJ5DJCOCHARK3VKZGROZCFT|null

**Reply by s-cork** [reply, marked as answer]
Ok try development dependency
include this in your native libs
`<script src="https://cdn.jsdelivr.net/npm/jquery-sparkline@2.4.0/jquery.sparkline.min.js"></script>`
And then the formatter should be
```python
from anvil.js.window import jQuery

...
    def sparkline_formatter(self, cell, on_rendered, **params):
        el = cell.getElement()
        def render():
            jQuery(el).sparkline(cell.getValue(), {"width":"100%", "type":"bar"})
        on_rendered(render)
```

**Reply by NicoN13** [reply]
![image](https://github.com/s-cork/Tabulator/assets/117717063/cea4c095-d57d-4b12-a0b8-56f43ed60558)

Once again, thanks @s-cork 
This is going to work brilliantly.

**Reply by s-cork** [reply]
fyi - that's now on the production branch or v2.2.15

**Reply by NicoN13** [reply]
I see params now contains a "on_rendered" key! Very nice.
